src/test/java/cat/udl/eps/softarch/demo/steps/CreateTakeStepDefs.java

```diff
@@ -0,0 +1,45 @@
+package cat.udl.eps.softarch.demo.steps;
+
+import cat.udl.eps.softarch.demo.domain.Take;
+import cat.udl.eps.softarch.demo.repository.TakeRepository;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+
+import io.cucumber.java.en.When;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+
+import java.math.BigDecimal;
+import java.time.ZonedDateTime;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+
+
+public class CreateTakeStepDefs {
+    public static String newResourceUri;
+    @Autowired
+    private StepDefs stepDefs;
+
+    @When("^I create a new Take with amount (\\d+), weight (\\d+) and location \"([^\"]*\")$")
+    public void iCreateANewTake(Integer amount, Integer weight, String location) throws Throwable {
+        Take take = new Take();
+        take.setAmount(amount);
+        take.setWeight(new BigDecimal(weight));
+        take.setLocation(location);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                post("/takes")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(stepDefs.mapper.writeValueAsString(take))
+                        .with(AuthenticationStepDefs.authenticate())
+        );
+        newResourceUri = stepDefs.result.andReturn().getResponse().getHeader("Location");
+    }
+
+    @And("^I can retrieve that Take$")
```

Comment:
If the meaning of the step is that you can retrieve that take because it exists, you can also include in the step implementation asserting that the response code is 200, instead of doing that in a separate step. 
Otherwise, if this step just tries to retrieve that take, and in a separate step the result code is checked, you might rephrase it `I try to retrieve that take`

Reply:
Fixed with my commit